### PR TITLE
Editor: Allow to re-open the current level

### PR DESCRIPTION
**Pull request by CyaNinix**
This is interesting when working in a team, when updating the level from a version control software.

### CODE REVIEW

**Comment by AMZN-daimini** [comment]
There is ongoing work to allow to hot reload prefabs when the source file is edited on disk, which should effectively achieve the same purpose. Tagging @o3de/sig-ui-ux for visibility and @EnochAmeyaw @srikappa-amzn for the prefab side.

Personally, I believe this could go in as a temporary aid and could be refactored once the other functionality is ready, if necessary.

**Comment by AMZN-daimini** [comment]
Also, @CyaNinix 
Commits to O3DE need to be signed to pass the DCO check. You may need to rebase and amend the commit to sign it retroactively. There's some info about this in the Checks tab of the PR :D

**Comment by AMZN-daimini** [comment]
> > Also, @CyaNinix Commits to O3DE need to be signed to pass the DCO check. You may need to rebase and amend the commit to sign it retroactively. There's some info about this in the Checks tab of the PR :D
> 
> Alright, sorry about that. I'll take care of that on monday on all the PR I opened.

No problem! It's a bit annoying at first, but then it becomes natural. Thank you for the contributions, they are very much appreciated :D

**Comment by CyaSid** [comment]
Hi @AMZN-daimini @srikappa-amzn ,

This is not quite the same as just having prefab hotreload.
We do some procedural level generation. When users want to try and generate different levels, it is always done in the same o3de map, which contains a couple entities necessary for the game to work (UI manager, game manager notably).
To ensure we cleanup everything properly, it is safer let the editor wipe the temporary entities by reloading the same level.

It is the same behavior as if the user wants to load another level: if an user has modifications pending, the editor default behavior when closing a level applies: a messagebox will prompt of existing unsaved change and the user still have the possibility to cancel.

This changeset uses code that have been around since CryEngine but was, for some reason, not used.

File: Code/Editor/Core/LevelEditorMenuHandler.cpp [hunk 1]
```diff
@@ -1058,7 +1058,7 @@ void LevelEditorMenuHandler::UpdateMRUFiles()
                 // Check file is still available
                 if (mruList->GetSize() > i)
                 {
-                    cryEdit->OpenDocumentFile((*mruList)[i].toUtf8().data());
+                    cryEdit->OpenDocumentFile((*mruList)[i].toUtf8().data(), true, COpenSameLevelOptions::ReopenLevelIfSame);
```
Comment (hultonha) [inline Code/Editor/Core/LevelEditorMenuHandler.cpp:1061]: suggestion: Would it be possible to please add a parameter comment for the `true` variable so it's clearer at the callsite to see what it's referring to (e.g. `, /*promptChanges=*/ true`)

**Review by gadams3 — APPROVED** [review]
Does this correctly handle prompting the user to save or discard any unsaved changes? As long as that's working properly, this seems really useful to me. Personally, I do a lot of iteration reusing and reloading the same level to undo all of my test or scripted changes and restore everything from the saved file. This is not the same as hot reloading when you want to discard unsaved changes.  Currently, this requires opening a different level, assuming there is one, then reopening the intended level. Being able to simply reopen the current level without opening another one or restarting the editor is a big time saver.

**Comment by CyaNinix** [comment]
> Does this correctly handle prompting the user to save or discard any unsaved changes?

I just tested it and yes it does.

**Comment by srikappa-amzn** [comment]
I agree with @gadams3  that reloading same level is useful if you just want to get back to a clean slate. I've seen it being used before. I think it makes sense to provide this option as well.

I've tested these changes locally and it looks like the prompt about unsaved changes only comes when using the 'Open Recent' option and doesn't come when reopening the same level with 'Open Level' option. While you are adding this, can you fix that too, to keep consistent behaviors for both options?

**Comment by srikappa-amzn** [comment]
Also pinging @rainbj as one of the UX expert around these areas to see whether he agrees with this change as well

**Comment by CyaNinix** [comment]
> I've tested these changes locally and it looks like the prompt about unsaved changes only comes when using the 'Open Recent' option and doesn't come when reopening the same level with 'Open Level' option. While you are adding this, can you fix that too, to keep consistent behaviors for both options?

I couldn't reproduce this issue, I got the prompt even with the "Open Level" option. However there was some missing code there, I fixed it.

**Review by srikappa-amzn — APPROVED** [review]

**Comment by srikappa-amzn** [comment]
@CyaNinix , the change fails in AR (Automated Review) run because of some assets failing processing. It's likely a product of not being on latest development. Can you fetch latest development and merge it into your branch and let us know? I can trigger another AR run then

**Pull request merged**